cf-revision-manager.php

```diff
@@ -2,200 +2,450 @@
 /*
 Plugin Name: CF Revision Manager
 Plugin URI: http://crowdfavorite.com
-Description: Revision management functionality so that plugins can add metadata to revisions as well as restore that metadata from revisions
-Version: 1.0.2
+Description: Revision management functionality so that plugins can add metadata to revisions as well as restore that metadata from revisions.
+Version: 2.0.0
 Author: Crowd Favorite
-Author URI: http://crowdfavorite.com 
-*/
-if (!class_exists('cf_revisions')) {
-	
-	define('CF_REVISIONS_DEBUG', false);
-	
-	function cfr_register_metadata($postmeta_key, $display_func = '') {
-		static $cfr;
-		if (empty($cfr)) {
-			$cfr = cf_revisions::get_instance();
-		} 
-		return $cfr->register($postmeta_key, $display_func);
-	}
+Author URI: https://crowdfavorite.com/
+License: GPL2
+License URI: https://www.gnu.org/licenses/gpl-2.0.html
+Domain Path: /languages
+Text Domain: cfrm
+ */
+
+// Exit if already accessed
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Register js and css file.
+ *
+ * @since 2.0.0
+ */
+function cfrm_enqueue_js() {
+	wp_register_script( 'cfrm-script', plugin_dir_url( __FILE__ ) . 'cfrm-script.js', array( 'jquery' ) );
+	wp_register_style( 'cfrm-style', plugin_dir_url( __FILE__ ) . 'cfrm-style.css' );
+}
+add_action( 'admin_enqueue_scripts', 'cfrm_enqueue_js' );
 
-	class cf_revisions {
-		private static $_instance;
-		protected $postmeta_keys = array();
-	
-		public function __construct() {
-			# save & restore
-			add_action('save_post', array($this, 'save_post_revision'), 10, 2);
-			add_action('wp_restore_post_revision', array($this, 'restore_post_revision'), 10, 2);
-
-			if (is_admin()) {		
-				# revision display
-				global $pagenow;
-				if ($pagenow == 'revision.php') {
-					add_filter('_wp_post_revision_fields', array($this, 'post_revision_fields'), 10, 1);
-					add_filter('_wp_post_revision_field_postmeta', array($this, 'post_revision_field'), 1, 2);
+/**
+ * Register revision manger admin menu.
+ *
+ * @since  2.0.0
+ *
+ * @return void
+ */
+function cfrm_admin_menu() {
+	 add_options_page(
+        __( 'Revision Manager', 'cfrm' ),
+        __( 'Revision Manager', 'cfrm' ),
+        'manage_options',
+        basename( __FILE__ ),
+        'cfrm_admin_form'
+    );
+}
+add_action( 'admin_menu', 'cfrm_admin_menu' );
+
+/**
+ * Revision Manger admin form callback function.
+ *
+ * @since  2.0.0
+ *
+ * @return void
+ */
+function cfrm_admin_form() {
+	//include_once plugin_dir_path( __FILE__ ) . 'includes/admin-form.php';
+	wp_enqueue_style( 'cfrm-style' );
+	wp_enqueue_script( 'cfrm-script' );
+	$required_keys = cfrm_set_required_keys();
+	$keys = array_diff( cfrm_get_distinct_meta_keys(), cfrm_excluded_keys(), $required_keys );
+?>
+<div class="wrap">
+	<h2><?php _e( 'CF Revision Manager', 'cfrm' ); ?></h2>
+<?php
+		if ( ! count( $keys ) ) {
```

Comment:
I would personally write this condition as `if ( empty( $keys ) ) {`. There's no performance or syntax reason for the change, it's purely a stylistic preference.